src/librustc/ty/error.rs

```diff
@@ -200,7 +200,9 @@ impl<'tcx> ty::TyS<'tcx> {
             ty::Array(_, n) => {
                 let n = tcx.lift_to_global(&n).unwrap();
                 match n.try_eval_usize(tcx, ty::ParamEnv::empty()) {
-                    Some(n) => format!("array of {} elements", n).into(),
+                    Some(n) => {
+                        format!("array of {} element{}", n, if n != 1 { "s" } else { "" }).into()
```

Comment:
This `if $n != 1 { "s" } else { "" }` business should be refactored to a function in some utility module in `rustc_errors`. Would be good to grep for `"s" } else { ""`.

Reply:
That would make a nice beginner issue. I'll open one.

Reply:
Opened https://github.com/rust-lang/rust/issues/64238.